jetty-servlet/src/test/java/org/eclipse/jetty/servlet/ErrorPageTest.java

```diff
@@ -198,6 +229,37 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
         }
     }
 
+    public static class AsyncSendErrorServlet extends HttpServlet implements Servlet
+    {
+        @Override
+        protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException
+        {
+            try
+            {
+                final CountDownLatch hold = new CountDownLatch(1);
+                AsyncContext async = request.startAsync();
+                async.start(() ->
+                {
+                    try
+                    {
+                        response.sendError(599);
+                        hold.countDown();
+                        async.complete();
```

Comment:
I'm not sure how this test is reliable?
Counting down the latch will put `complete()` in a race with the thread that is exiting `doGet()`, so the events won't be in a reliable order, no?

Reply:
Ah yes - that aspect of the test is now a race.   When I wrote the test, sendError wasn't async (unless the target resource was).
The main focus of the test is to ensure we don't renter as we once did, so that is tested.  We also don't really notice if the complete does ISE or not.... so I'll see if I can modify the test so that we test both scenarios of the complete winning and losing the race....

Reply:
I have split that test into two and it has revealed that we have a bigger problem.  So this concern is resolved, but one of the tests is now failing.